Render religious landuse and place of worship lighter

jeisenbe opened the pull request:
Fixes #3457 
Fixes #3322

Changes proposed in this pull request:

- Change the fill color for religious landuse and places of worship to a lighter shade of gray

- Change the outline to 30% darker than the new shade; significantly lighter than the previous outline

Test rendering with links to the example places:

**Wamena, Indonesia**
http://openstreetmap.org/#map=17/-4.09339/138.94656
z15 Before:
![](https://user-images.githubusercontent.com/42757252/47572515-1ff02400-d976-11e8-9eef-3d42d9a318e4.png)

After:
![](https://user-images.githubusercontent.com/42757252/47572550-34342100-d976-11e8-92ca-aa2e015e0355.png)

z16 Before
![](https://user-images.githubusercontent.com/42757252/47568613-8a9c6200-d96c-11e8-87bb-1d02f54b5f84.png)

After:
![](https://user-images.githubusercontent.com/42757252/47568124-40ff4780-d96b-11e8-8a4c-e9d81e717fc9.png)

z17 Before
![](https://user-images.githubusercontent.com/42757252/47568606-86704480-d96c-11e8-9053-f1409c656ed8.png)

After:
![](https://user-images.githubusercontent.com/42757252/47568113-393fa300-d96b-11e8-9060-746cfdff09ae.png)


**YWAM campus in Kona, Hawaii**
(This is a large religious campus)
http://openstreetmap.org/#map=16/19.6287/-155.9750

z15 Before:
![](https://user-images.githubusercontent.com/42757252/47598273-c8cd6c00-d9d4-11e8-8f59-3bc394928948.png)

After:
![](https://user-images.githubusercontent.com/42757252/47598282-daaf0f00-d9d4-11e8-8a9f-8a515a3d775f.png)

z16 Before:
![ywamz16original](https://user-images.githubusercontent.com/42757252/47923960-13cc0f80-defe-11e8-976c-ad5b452542aa.png)

After:
![](https://user-images.githubusercontent.com/42757252/47598283-dedb2c80-d9d4-11e8-86d2-cf2ebb4d9f37.png)

kocio-pl commented:
Did you try lighter alternatives? I guess that it could be still less dark.

jeisenbe commented:
Yeah, I don’t know why it is showing all of the previous commits in my
fork, even though I just created this branch last week. Should I rebase the
branch on master or will that cause more problems? Or squash the commits?
On Sat, Nov 3, 2018 at 6:26 AM polarbearing <notifications@github.com>
wrote:

> The commit list is very long, with unrelated items, for changing just two
> colour values?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/gravitystorm/openstreetmap-carto/pull/3493#issuecomment-435514023>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AoxshH-9ju8DxAYZ2fK0uy0AcbMGhzwkks5urLhvgaJpZM4YL2db>
> .
>

jeisenbe commented:
Sorry for the delay, I meant to post these but my computer crashed.
Here are some renderings with #d6d6d6
I also tried lighter shades of gray, but I believe they are too close to residential.
This is about the limit, the lightest shade of gray that is still distinguishable:

z16 Wamena
![religious-z16](https://user-images.githubusercontent.com/42757252/48065153-3ebfa780-e20d-11e8-9150-1349b001c0e8.png)

z17 Wamena
![religious-z17](https://user-images.githubusercontent.com/42757252/48065155-3f583e00-e20d-11e8-9e2b-5f4c0b77e1f0.png)

z14 Kona
![ywam-z14](https://user-images.githubusercontent.com/42757252/48065161-45e6b580-e20d-11e8-9d19-239f628bd783.png)

z15 Kona
![ywam-z15](https://user-images.githubusercontent.com/42757252/48065165-4a12d300-e20d-11e8-9a54-e9aefae90598.png)

z16 Kona
![ywam-z16](https://user-images.githubusercontent.com/42757252/48065173-50a14a80-e20d-11e8-90c3-adac03a34905.png)

kocio-pl commented:
I guess this is good enough for me. Any other opinons?

jragusa commented:
What about name rendering? For example, a lot of of these surfaces contain a name at Milano (see [overpass request](https://overpass-turbo.eu/s/Dr6))

kocio-pl commented:
Also this Kona campus has name:

https://www.openstreetmap.org/way/101222222

It makes a lot of sense, on the other hand it should be tested if such names do not lead to clutter - I mean eclipsing symbols and names of churches/temples.

kocio-pl commented:
BTW: `denomination=foursquare`? :smile: 

https://www.openstreetmap.org/way/607212018

jeisenbe commented:
Good idea, I’ll try adding name labels.

(Foursquare, officially “The International Church of the Foursquare Gospel,
is a evangelical/protestant Christian denomination from the
Pentecostal/Charismatic tradition, founded by Amy Semple-McPherson in Los
Angeles in 1924; they “preach the foursquare Gospel.” Their logo has 4
symbols: a cross, crown, cup and dove. There is a big church in LA called
“Angeles Temple which used to be famous for extravagant Christmas pageants
and passion plays with live animals and flying “angels” thanks to
congregation members who worked in the Hollywood film business when Mrs.
Semple-McPherson was the preacher.

Probably they should have defended that trademark better...)
On Wed, Nov 7, 2018 at 1:20 AM kocio-pl <notifications@github.com> wrote:

> BTW: denomination=foursquare? 😄
>
> https://www.openstreetmap.org/way/607212018
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/gravitystorm/openstreetmap-carto/pull/3493#issuecomment-436313142>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AoxshJ5X2rloGY3FuvLcmiTrpXHxp7DNks5usba3gaJpZM4YL2db>
> .
>

jeisenbe commented:
Here are two options. 

**1) Add landuse_religious to the current rendering for amenity_place_of_worship,** 
plus a new line to allow lower zoom levels with waypixels >3000, as with other similar features

Kona z14
![kona-z14-standard](https://user-images.githubusercontent.com/42757252/48132267-557e0100-e2d6-11e8-9a51-2b450a86aade.png)

Kona z15
![kona-z16-standard](https://user-images.githubusercontent.com/42757252/48131766-8fe69e80-e2d4-11e8-9e08-3ccde71929c7.png)

Kona z16
![kona-z16-standard](https://user-images.githubusercontent.com/42757252/48131763-8d844480-e2d4-11e8-812c-61eca8874ce5.png)

**2) Change rendering to match that of other landuses and societal amenities**

Kona z14
![kona-z14](https://user-images.githubusercontent.com/42757252/48132067-94f81d80-e2d5-11e8-8400-862e3c3e8b58.png)

Kona z15
![kona-z15](https://user-images.githubusercontent.com/42757252/48132065-945f8700-e2d5-11e8-8fed-b98aa726b94b.png)

Kona z16
![kona-z16](https://user-images.githubusercontent.com/42757252/48132053-8dd10f80-e2d5-11e8-9138-7e1a8745658d.png)


**Code 1):**
```
  [feature = 'amenity_place_of_worship'],
  [feature = 'landuse_religious'] {
    [zoom >= 10][way_pixels > 3000][is_building = 'no'],
    [zoom >= 17] {
      text-name: "[name]";
      text-size: @standard-font-size;
      text-wrap-width: @standard-wrap-width;
      text-line-spacing: @standard-line-spacing-size;
      text-fill: #000033;
      text-dy: 12;
      text-face-name: @standard-font;
      text-halo-radius: @standard-halo-radius;
      text-halo-fill: @standard-halo-fill;
      text-placement: interior;
    }
  }
```
**Code 2):**

```
  [feature = 'amenity_place_of_worship'],
  [feature = 'landuse_religious'] {
    [zoom >= 10][way_pixels > 3000][is_building = 'no'],
    [zoom >= 17] {
      text-name: "[name]";
      text-size: @landcover-font-size;
      text-wrap-width: @landcover-wrap-width-size;
      text-line-spacing: @landcover-line-spacing-size;
      text-fill: #333333;
      text-dy: 12;
      [way_pixels > 12000] {
        text-size: @landcover-font-size-big;
        text-wrap-width: @landcover-wrap-width-size-big;
        text-line-spacing: @landcover-line-spacing-size-big;
      }
      [way_pixels > 48000] {
        text-size: @landcover-font-size-bigger;
        text-wrap-width: @landcover-wrap-width-size-bigger;
        text-line-spacing: @landcover-line-spacing-size-bigger;
      }
      text-face-name: @landcover-face-name;
      text-halo-radius: @standard-halo-radius;
      text-halo-fill: @standard-halo-fill;
      text-placement: interior;
    }
  }
```

kocio-pl commented:
It would be interesting to see how it works with churches and other temples visible currently.

kocio-pl commented:
Kona campus should be tagged as `amenity=university` by the way.

jeisenbe commented:
Hope Chapel
1)Standard code
![hopechapel-z16-standard](https://user-images.githubusercontent.com/42757252/48143859-101bfc80-e2f3-11e8-9f02-72f8677c4f44.png)
2)Landuse code
![hopechapel-z16-landuse](https://user-images.githubusercontent.com/42757252/48143838-01cde080-e2f3-11e8-99a3-427965a2bce2.png)

Hawaiian Place of Worship (temple ruins)
1) Standard 
![hawaiian-pow-z17-standard](https://user-images.githubusercontent.com/42757252/48143864-1316ed00-e2f3-11e8-8554-851330bebbbf.png)
2) Landuse
![hawaiian-pow-z17-landuse](https://user-images.githubusercontent.com/42757252/48143846-07c3c180-e2f3-11e8-8f7c-2ced39aa7817.png)

Bhuddist POW
1) Standard
![mission-z17-standard](https://user-images.githubusercontent.com/42757252/48144441-39895800-e2f4-11e8-8424-8c02ba3d07cf.png)

2) Landuse
![mission-z17-landuse](https://user-images.githubusercontent.com/42757252/48143855-0c887580-e2f3-11e8-9935-8c4ee009d806.png)

jeisenbe commented:
So I need to find a named landuse=religious polygon with multiple smaller
place_of_worship areas or nodes inside with different names.

How can I search for that?

Or I could invent one on JOSM
On Thu, Nov 8, 2018 at 1:21 AM kocio-pl <notifications@github.com> wrote:

> I meant checking the situation when both temple and landuse have their
> respective names. Won't the name of the landuse eclipse name+icon of a
> temple?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/gravitystorm/openstreetmap-carto/pull/3493#issuecomment-436682626>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AoxshO5nCWGZ25XBQtHeNm-vKER7a2oaks5uswiHgaJpZM4YL2db>
> .
>

jeisenbe commented:
Fixes #3322 

Amenity=place_of_worship labels will continue to be rendered the same as currently.

Landuse=religious name label will be rendered the same as other landuse labels.
 
I've done the renderings with #d0d0d0 as the color for the area, and darkened it 50% for the religious landuse label.

I added the names to some areas in my town on JOSM for testing

z17 Landuse=religious with 2 place_of_worship buildings and a node, in town.
![z17-gkip-new-png](https://user-images.githubusercontent.com/42757252/48201492-ab6daa00-e3a5-11e8-951a-cb471091ce9c.png)

z18
![z18-gkip-new](https://user-images.githubusercontent.com/42757252/48201496-ac9ed700-e3a5-11e8-8598-ab907e648537.png)

z19
![z19-gkip-new](https://user-images.githubusercontent.com/42757252/48201497-ad376d80-e3a5-11e8-811a-f867088496a2.png)

Church in the country (it's a round building)
z17
![z17-katolik-new](https://user-images.githubusercontent.com/42757252/48201503-b45e7b80-e3a5-11e8-83d9-7aec0293e084.png)

z18
![z18-katolik-new](https://user-images.githubusercontent.com/42757252/48201507-b6283f00-e3a5-11e8-8753-8bcb40f81336.png)

Test with large religious area with place_of_worship building almost at the center, made to see how the conflicting labels would resolve. It behaves as expected.
z15
![z15-kecandian-new](https://user-images.githubusercontent.com/42757252/48201512-bc1e2000-e3a5-11e8-858f-9338aac067df.png)

z16
![z16-kecandian-new](https://user-images.githubusercontent.com/42757252/48201518-c04a3d80-e3a5-11e8-92ef-41dd2a66ab34.png)

z17
![z17-kecandian-new](https://user-images.githubusercontent.com/42757252/48201526-c2ac9780-e3a5-11e8-92ea-f87653f47946.png)

z18
![z18-kecandian-new](https://user-images.githubusercontent.com/42757252/48201531-c6401e80-e3a5-11e8-9f3e-b6a4a0f328d0.png)

kocio-pl commented on amenity-points.mss:
```diff
@@ -1808,12 +1808,37 @@
     text-wrap-width: @standard-wrap-width;
     text-line-spacing: @standard-line-spacing-size;
-    text-fill: #000033;
+    text-fill: #000033; 
```
Just a space to remove at the end.

jeisenbe commented:
It should be fixed now
On Thu, Nov 8, 2018 at 11:12 PM kocio-pl <notifications@github.com> wrote:

> *@kocio-pl* commented on this pull request.
> ------------------------------
>
> In amenity-points.mss
> <https://github.com/gravitystorm/openstreetmap-carto/pull/3493#discussion_r231900882>
> :
>
> >    [feature = 'amenity_place_of_worship'][zoom >= 17] {
>      text-name: "[name]";
>      text-size: @standard-font-size;
>      text-wrap-width: @standard-wrap-width;
>      text-line-spacing: @standard-line-spacing-size;
> -    text-fill: #000033;
> +    text-fill: #000033;
>
> Just a space to remove at the end.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/gravitystorm/openstreetmap-carto/pull/3493#pullrequestreview-172973510>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AoxshBBRf79PgvAkJneGYmbUISEIO1cgks5utDu8gaJpZM4YL2db>
> .
>

kocio-pl commented:
I'm happy with the results. Please just remove the extra space you have introduced by accident and this will be ready to merge - see https://github.com/gravitystorm/openstreetmap-carto/pull/3493#discussion_r231900882.

polarbearing commented:
yes d0d0d0 is fine for me

jeisenbe commented:
Should be fixed now.

It’s strange; I fixed it locally on my system and used $ git push, and it
says all the changes are pushed, but github didn’t show the file change.

So I fixed it on GitHub.

Please comment on the new github workflow issue if you have time.
On Sat, Nov 10, 2018 at 8:05 AM polarbearing <notifications@github.com>
wrote:

> yes d0d0d0 is fine for me
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/gravitystorm/openstreetmap-carto/pull/3493#issuecomment-437523305>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AoxshHfvFhKl2mXrNe7hovvF-hYmYEn7ks5utgpUgaJpZM4YL2db>
> .
>

kocio-pl commented:
Thanks for the change and adding the label (thanks, @polarbearing!).

I'm not able to give more attention to the OSM Carto than I already do, sorry about that. I'm not sure what was the reason and it takes some time to investigate, but in the end the most important thing is that I can squash and merge your code properly.
